plugin/src/ServeSession.lua

```diff
@@ -207,6 +247,162 @@ function ServeSession:__onActiveScriptChanged(activeScript)
 	self.__apiContext:open(scriptId)
 end
 
+function ServeSession:__replaceInstances(idList)
+	if #idList == 0 then
+		return true, PatchSet.newEmpty()
+	end
+	-- It would be annoying if selection went away, so we try to preserve it.
+	local selection = Selection:Get()
+	local selectionMap = {}
+	for i, instance in selection do
+		selectionMap[instance] = i
+	end
+
+	-- TODO: Should we do this in multiple requests so we can more granularly mark failures?
+	local modelSuccess, replacements = self.__apiContext
+		:serialize(idList)
+		:andThen(function(response)
+			Log.debug("Deserializing results from model endpoint")
+			local objects = SerializationService:DeserializeInstancesAsync(response.modelContents)
+			local instanceMap = {}
+			for _, item in objects[1]:GetChildren() do
```

Comment:
Tiny concern here: `objects[1]` is never checked to be non nil and I don't trust that it won't somehow be empty

Reply:
I guess I've never really considered what would happen if you tried to deserialize an empty model using `SerializationService` but this is a valid concern. Should probably add a check to make sure that all of the requested objects actually got delivered in case something silently fails.